modules/docs/lib/widgetUtils.tsx

```diff
@@ -0,0 +1,299 @@
+/**
+ * React context to track the current heading level. This is useful to relatively change heading
+ * levels of a subtree
+ */
+export const HeadingLevelContext = React.createContext(3);
+
+/**
+ * React context to keep track of the current indent level
+ */
+export const IndentLevelContext = React.createContext(0);
+
+/**
+ * Utility function for getting non-breaking space for code formatting. It will print a string of
+ * non-breaking spaces according to the level provided. These characters are safe for a React render
+ * function.
+ */
+export function space(level: number) {
+  return [...Array(level * 2)].map(v => '\u00A0').join('');
+}
+
+export function capitalize(input: string) {
+  return input.charAt(0).toUpperCase() + input.slice(1);
+}
+
+interface HeadingProps {
+  headingOffset?: number;
+}
+/**
+ * Special heading component that uses @mdx-js/react heading components, but is also aware of
+ * relative heading levels.
+ */
+export const Heading = createComponent('h4')({
+  displayName: 'MDXHeading',
+  Component({headingOffset = 0, ...elemProps}: HeadingProps) {
+    const headingLevel = React.useContext(HeadingLevelContext);
+    const as = `h${headingLevel + headingOffset}` as 'h4'; // Make Typescript happy
+    return <MDX as={as} {...elemProps} />;
+  },
+});
+
+export function renderTypeParameters(typeParameters?: types.TypeParameter[]) {
+  return typeParameters && typeParameters.length ? (
+    <RenderContext.Provider value="inline">
+      <span>&lt;</span>
+      {typeParameters.map((p, index) => {
+        return (
+          <React.Fragment key={index}>
+            {index !== 0 && <span className="token punctuation">, </span>}
+            <Value value={p} />
+          </React.Fragment>
+        );
+      })}
+      <span>&gt;</span>
+    </RenderContext.Provider>
+  ) : (
+    ''
+  );
+}
+
+const ButtonHyperLink = Hyperlink.as('button');
+
+export interface SymbolDialogProps {
+  value: types.SymbolValue;
+  hideDescription?: boolean;
+}
+
+export const SymbolDialog = ({value, hideDescription}: SymbolDialogProps) => {
+  const [symbol, setSymbol] = React.useState<types.ExportedSymbol | undefined>(undefined);
+  const model = useDialogModel({
+    onShow() {
+      setSymbol(docs.find(d => value.name === d.name) || undefined);
+    },
+  });
+
+  return (
+    <Dialog model={model}>
+      <>
+        <Dialog.Target
+          as={ButtonHyperLink}
+          className="token symbol"
+          style={{
+            border: 'none',
+            background: 'transparent',
+            fontSize: 'inherit',
+            fontFamily: 'inherit',
+          }}
+          aria-haspopup="true"
+        >
+          {value.displayName || value.name}
+        </Dialog.Target>
+        {renderTypeParameters(value.typeParameters)}
+      </>
+      <Dialog.Popper>
+        <Dialog.Card maxHeight="50vh" maxWidth="90vh">
+          <Dialog.CloseIcon />
+          <Dialog.Heading>
+            <>
```

Comment:
i think we can drop the fragment here if we also drop the commented line two lines down from here

Reply:
or at least, drop that comment, even if we still need a fragment, but i think the fragment will go away too